core/lib/types/src/api/mod.rs

```diff
@@ -19,6 +19,39 @@ use crate::{
 
 pub mod en;
 
+/// Enum for choosing API mode
+#[derive(Copy, Clone, Debug, PartialEq, Display, Default)]
+pub enum ApiMode {
+    #[default]
+    Modern,
+    EthTransferIncluded,
```

Comment:
I'm not sure if the names respect well the intent. Modern doesn't say much to someone who is familiar with the topic already. Probably `EthTransferFiltered` and `EthTransferIncluded` would be better.

Reply:
Actually, 1) I do not think that we're going to support more modes for our API, and 2) this is a temporary measure until we implement a proper protocol upgrade, so maybe it's more generic than needed.
Maybe it'll be even better to have an enum named `ApiEthTransferEvents` and values `Enabled` and `Disabled`.
This way this will be an exhaustive list of supported options, and you won't have to specify that "Currently we support these modes", as there may be only these 2.